code/modules/clothing/clothing.dm

```diff
@@ -292,6 +291,30 @@
 			siemens_coefficient = between(0, material.conductivity / 10, 10)
 		slowdown = between(0, round(material.weight / 10, 0.1), 6)
 
+/obj/item/clothing/proc/get_accessory(var/typepath)
+	if(istype(src, typepath))
+		return src
+	if(LAZYLEN(accessories))
+		var/accessory = locate(typepath) in accessories
+		if(accessory)
+			return accessory
+	return null
+
+/obj/item/clothing/Destroy()
+	SEND_SIGNAL(src, COMSIG_PARENT_QDELETING, args)
```

Comment:
this signal should already be sent on previous destroy calls, not just clothing calls, otherwise it isn't a good generic signal